### PR TITLE
optimization by avoiding loops for regression over numpy array.

### PR DESCRIPTION
**Pull request by bpinsard**
## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [ ] I've linked relevant issues in the PR body
- [ ] I've applied the relevant labels to this PR
- [ ] I've added relevant tests for my contribution
- [ ] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer

## Description

The implementation was using loops to perform the same linear regression, with same regressors on a 4d array. Significant speedup is noticeable in profiling which should help for near-rt application.

### CODE REVIEW

**Review by po09i — APPROVED** [review]
Looks good! thanks for the improvement!

**Review by po09i — DISMISSED** [review]
Just noticed this was still a WIP, it looks good to me for now, if you were done with the PR let me know by switching to ready for review!

File: shimmingtoolbox/shim/realtime_zshim.py [hunk 1]
```diff
@@ -131,27 +131,10 @@ def realtime_zshim(nii_fieldmap, nii_anat, pmu, json_fmap, nii_mask_anat_riro=No
...
-    for g_axis in range(3):
-        for i_x in range(fieldmap.shape[0]):
-            for i_y in range(fieldmap.shape[1]):
-                for i_z in range(fieldmap.shape[2]):
-
-                    # do regression to separate static component and RIRO component
-                    reg = LinearRegression().fit(acq_pressures.reshape(-1, 1) - mean_p,
-                                                 -gradient[g_axis][i_x, i_y, i_z, :])
-                    # Multiplying by the RMS of the pressure allows to make abstraction of the tightness of the bellow
-                    # between scans. This allows to compare results between scans.
```
Comment (gaspardcereza) [inline shimmingtoolbox/shim/realtime_zshim.py:151]: This comment is useful, we should keep it

**Comment by gaspardcereza** [comment]
I haven't checked if the returned `riro` and `static` are exactly the same as before but if they are indeed, I like this implementation which is way clearer ! Thanks @bpinsard

**Review by gaspardcereza — APPROVED** [review]
I checked the shim results and they're unaffected by these changes. Awesome improvement thank you very much @bpinsard !

**Pull request merged**